lib/assessment.js

```diff
@@ -29,7 +43,11 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  var ary;
+  for (var i = 0; i < length; i++){
+    ary[i] = defaultsFunction;
+  }
+  return ary; //i think i am unclear on what line 44 is asking - same thing as Q3 but with a function??
```

Comment:
It's saying we should be adding `defaultFunction(i)` to the array each time - invoking `defaultFunction` with the index as the function's input value.